src/components/Button/Button.js

```diff
@@ -18,36 +19,61 @@ const { defaultProps, propTypes } = CarbonButton;
 
 // TODO: V3 - Remove deprecated props `largeText`.
 
-const Button = ({ className, largeText, size, kind, ...other }) => {
+const Button = ({
+  className,
+  disabled,
+  largeText,
+  loading,
+  kind,
+  renderIcon,
+  size,
+  ...other
+}) => {
   const isSize = value => size === value;
   const isLarge = isSize('large') || largeText || isSize('lg') || isSize('xlg');
-
   const isGhostDanger = kind === 'ghost-danger';
 
+  const buttonClasses = classnames(namespace, className, {
+    [`${namespace}--ghost-danger`]: isGhostDanger,
+    [`${namespace}--large`]: isLarge,
+    [`${namespace}--loading`]: loading,
+  });
+
   return (
     <CarbonButton
-      className={classnames(namespace, className, {
-        [`${namespace}--large`]: isLarge,
-        [`${namespace}--ghost-danger`]: isGhostDanger,
-      })}
+      className={buttonClasses}
+      disabled={disabled || loading}
+      kind={isGhostDanger || loading ? 'ghost' : kind}
+      renderIcon={loading ? InlineLoading : renderIcon}
```

Comment:
Yeah exactly!

For icons, there isn't meant to be a loading spinner alongside a regular icon... so the spinner needs to replace an extra icon if provided.

For the `disabled` state, I applied that because it seemed to make sense. 🤔 If the element is not meant to be interacted with yet as a true button, and it's not a skeleton component, then seems best to set it as `disabled`

Reply:
Also, the logic for `kind` is such that if `loading={true}`, then it is a `ghost` button. The reasoning for this is that it looked weird to have the loading spinner on a `primary` button. 🤔 And every mockup I've seen shows a `ghost` button with the spinner. But I'm happy to adjust this! A `secondary` button would also look fine with the spinner...